src/api.v2/model/Experiment.js

```diff
@@ -68,17 +69,26 @@ class Experiment extends BasicModel {
       .leftJoin(`${tableNames.EXPERIMENT_PARENT} as p`, 'e.id', 'p.experiment_id')
       .as('mainQuery');
 
-    const result = await collapseKeyIntoArray(
+    const experiments = await collapseKeyIntoArray(
       mainQuery,
       [...fields, 'parent_experiment_id', 'is_subsetted'],
       'key',
       'metadataKeys',
       this.sql,
     );
 
-    return result;
+    // Assuming the getMetadataByExperimentId function is async and returns the metadata
+    const cellLevel = new CellLevel();
+
+    await Promise.all(experiments.map(async (experiment) => {
+      const result = await cellLevel.getMetadataByExperimentId(experiment.id);
+      experiment.cellLevelMetadata = result;
+    }));
```

Comment:
Why not get all the experiment's metadata at once? In the admin for example it's the difference between doing > 2000 queries vs only one.

You don't need to change the function `getMetadataByExperimentId` much, just pass as a param to this func the array of exp ids.

Then in the `where` map the array of experiment ids to `[{experiment_id: ...}, {experiment_id: ...}, ...]`

Reply:
oh yeah, nice